### DISCUSSION TITLE
LevelSetMeshRefinementTransfer do not transfer with initial_steps != 0

### DISCUSSION THREAD

**Discussion by doquang** [idaholab/moose#23533] [Q&A General]
I am having a problem when transferring adaptivity information that contain an information of 'initial_steps':
```*** ERROR ***
The following error occurred in the object "marker_to_sub", of type "LevelSetMeshRefinementTransfer".
The meshes must be identical to utilize MultiAppDofCopyTransfer::transfer.
```

It seems the LevelSetMeshRefinementTransfer do not support it yet. My code will run without the setting of 'initial_steps', or 'initial_steps=0'
Here is the input from parent.i 
```
[Adaptivity]
  initial_marker = markers
  initial_steps = 2
  max_h_level = 2
  cycles_per_step = 2
  marker = markers
  ...

[MultiApps]
  [sub]
    type = TransientMultiApp
    input_files = 'sub.i'
    execute_on = TIMESTEP_END
  []
[]
[Transfers]
  [marker_to_sub]
    type = LevelSetMeshRefinementTransfer
    to_multi_app = sub
    source_variable = markers
    variable = markers
    check_multiapp_execute_on = false
  []
```

**Comment by jiangwen84** [comment]
I am not able to reproduce your error. 

I added ```initial_steps=2``` in the ```modules/level_set/test/tests/transfers/markers/multi_level/parent.i```. The modified input file was ran without any error.

**Reply by doquang** [reply]
Hi @jiangwen84 ,
Please use this input file instead of `modules/level_set/test/tests/transfers/markers/multi_level/parent.i`:
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 10
  ny = 10
  uniform_refine = 2
[]

[Adaptivity]
  initial_marker = init_marker
  initial_steps = 2
  marker = marker
  max_h_level = 2
  cycles_per_step = 2
  [./Indicators]
    [./error]
      type = GradientJumpIndicator
      variable = u
    [../]
  [../]
  [./Markers]
    [./marker]
      type = ErrorFractionMarker
      coarsen = 0.4
      refine = 0.5
      indicator = error
    [../]
    [init_marker]
      type = BoxMarker
      bottom_left = '0.35 0.25 0'
      top_right = '0.5 0.5 0'
      inside = refine
      outside = coarsen
    []
  [../]
[]

[Variables]
  [./u]
  [../]
[]

[Kernels]
  [./time]
    type = TimeDerivative
    variable = u
  [../]
  [./diff]
    type = Diffusion
    variable = u
  [../]
[]

[BCs]
  [./left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  [../]
  [./right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  [../]
[]

[Problem]
  type = LevelSetProblem
[]

[Executioner]
  type = Transient
  dt = 0.02
  num_steps = 4
  solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]

[MultiApps]
  [./sub]
    type = TransientMultiApp
    input_files = 'sub.i'
    execute_on = TIMESTEP_END
  [../]
[]

[Transfers]
  [./marker_to_sub]
    type = LevelSetMeshRefinementTransfer
    to_multi_app = sub
    source_variable = marker
    variable = marker
    check_multiapp_execute_on = false
  [../]
[]

[Outputs]
  hide = u
  exodus = true
[]
```

**Reply by jiangwen84** [reply]
Ok, I see. 

As you can see from your input file, the ```marker``` is transferred to the sub app through ```LevelSetMeshRefinementTransfer```. We have not implemented the code to transfer ```initial_marker```, so the sub app won't do the initial adaptivity. It should be straightforward to add this capability by following what has been done in   ```LevelSetMeshRefinementTransfer```.

Do you have to use initial adaptivity? I think you can work around it if you do not want to write new codes.

**Reply by doquang** [reply]
With an initial adaptivity, variables will be initialized with good resolution. Without it, all variables will start with coarse resolution in the first few steps.

**Reply by jiangwen84** [reply]
> With an initial adaptivity, variables will be initialized with good resolution. Without it, all variables will start with coarse resolution in the first few steps.

You can probably use higher level uniform refinement, i.e. uniform_refine = 4, for your initial conditions. As level set evolves, the mesh can be coarsen or refined just near the level set region.
